### ISSUE TITLE
Migrate Travis CI checks to GitHub workflows

### ISSUE THREAD

**Issue by TysonAndre** [microsoft/tolerant-php-parser#407]
https://app.travis-ci.com/github/Microsoft/tolerant-php-parser/builds is apparently a blank page and broken image at the time this issue was created.

This seems to be why https://github.com/microsoft/tolerant-php-parser/pull/405/files was created

The same functionality is possible (e.g. also run it in docker similar to how phpunit tests are run in GH Actions) if configuration is added (and tested)

**Comment by roblourens** [comment]
You started this awhile back in https://github.com/microsoft/tolerant-php-parser/pull/386, should I merge that?

**Comment by TysonAndre** [comment]
> You started this awhile back in https://github.com/microsoft/tolerant-php-parser/pull/386, should I merge that?

Yes. PHPStan may have had some updates that would cause failures due to dead code (seen locally), which #406 would fix.
